crates/ruff_server/src/session/settings.rs

```diff
@@ -418,6 +414,37 @@ impl ResolvedClientSettings {
                     ConfigurationPreference::EditorFirst,
                 ),
             },
+        };
+
+        if contains_invalid_settings {
+            show_err_msg!(
+                "Ruff received invalid client settings. Refer to the logs for more information."
```

Comment:
As a user (mainly VS Code), I'd find it difficult to understand what client settings are because I never configured them explicitly. Can we say invalid Ruff extension settings or invalid Ruff LSP settings?

Reply:
We can say "Ruff received invalid settings from the editor. ..." to make it not specific to VS Code and LSP (implementation detail).